### PR TITLE
Detect composting

### PR DESCRIPTION
**Pull request by MattAKAFred**
Detects interactions with composter block

### CODE REVIEW

File: src/me/ryanhamshire/AutomaticInventory/AIEventHandler.java [hunk 1]
```diff
@@ -97,6 +97,16 @@ public void onFeedAnimal(PlayerInteractEntityEvent event)
...
         tryRefillStackInHand(player, event.getHand());
     }
+    
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+    public void onCompostItem(PlayerInteractEvent event)
+    {
+        Player player = event.getPlayer();
+        Block clickedBlock = event.getClickedBlock();
+        if(clickedBlock == null) return;
+        if (!(clickedBlock.getType() == Material.COMPOSTER)) return;
```
Comment (RoboMWM) [inline src/me/ryanhamshire/AutomaticInventory/AIEventHandler.java]: `!=` would be clearer?

**Review by RoboMWM — COMMENTED** [review]
I'm pretty sure you could add this material check to one of the already-existing PlayerInteractEvent handlers.

**Comment by MattAKAFred** [comment]
> I'm pretty sure you could add this material check to one of the already-existing PlayerInteractEvent handlers.

The only other interact event handler I see is `PlayerInteractEntityEvent` - unless I'm missing something?

File: src/me/ryanhamshire/AutomaticInventory/AIEventHandler.java (hunk 1 above)
Comment (MattAKAFred) [reply src/me/ryanhamshire/AutomaticInventory/AIEventHandler.java]: Fixed in commit

**Comment by RoboMWM** [comment]
I'll admit I took a blind guess. If there is no such event handler then that's ok

I do wonder if a new event was created for that action tho? At quick glance I don't see anything the javadocs to suggest that tho. Hmm. Well, if none exists, maybe you can check if the action was `RIGHT_CLICK_BLOCK` since I assume left clicks wouldn't do anything? (This would be to reduce the number of unnecessary getType calls, which I've heard can be expensive - and playerinteractevent is a commonly-called event.)

**Review by RoboMWM — APPROVED** [review]

**Pull request merged**